client/src/components/TodoList.vue

```diff
@@ -9,27 +9,53 @@
       </div>
     </div>
      <!-- // we are now passing the data to the todo component to render the todo list -->
-    <todo v-on:delete-todo="deleteTodo" v-on:toggle-todo="toggleTodo" v-for="todo in todos" v-bind:todo="todo" :key="todo.id">></todo>
+    <todo v-on:delete-todo="deleteTodo" v-on:toggle-todo="toggleTodo" v-on:edit-todo="editTodo" v-for="todo in todos" v-bind:todo="todo" :key="todo.id">></todo>
   </div>
 </template>
 
 <script type = "text/javascript" >
-
+import axios from 'axios';
 import Todo from './Todo';
 
+const url = 'http://localhost:9393';
+
 export default {
   props: ['todos'],
   components: {
     Todo,
   },
   methods: {
     deleteTodo(todo) {
-      const todoIndex = this.todos.indexOf(todo);
-      this.todos.splice(todoIndex, 1);
+      axios.delete(`${url}/todo/${todo.id}`)
+      .then((response) => {
+        if (response.data.success === 'ok') {
+          const todoIndex = this.todos.indexOf(todo);
+          this.todos.splice(todoIndex, 1);
+        }
+      })
+      .catch((error) => {
+        this.status = `an error ocurred: ${error}`;
+      });
     },
     toggleTodo(todo) {
-      const todoIndex = this.todos.indexOf(todo);
-      this.todos[todoIndex].done = !this.todos[todoIndex].done;
+      const todoToggle = todo;
+      todoToggle.done = !todo.done;
+      axios.patch(`${url}/todo/${todo.id}`, todoToggle)
+      .then(() => {
+        //
+      })
+      .catch((error) => {
+        this.status = `an error ocurred: ${error}`;
+      });
+    },
+    editTodo(todo) {
```

Comment:
It's better to keep the server update logic to the single todo component, as it's the responsability of it

Reply:
I just maintained the logic of the original to-do Vue tutorial. The edit, toggle and delete functions were all implemented on the TodoList component

Reply:
ok